src/frontend/src/observer/observer_manager.rs

```diff
@@ -86,16 +87,16 @@ impl ObserverState for FrontendObserverNode {
         match resp.info {
             Some(Info::Snapshot(snapshot)) => {
                 for db in snapshot.databases {
-                    catalog_guard.create_database(db)
```

Comment:
We should also create all views when handling initialization notification. The failure in deterministic batch recovery test is possibly cause by it:
```
deterministic simulation e2e, ci-3cn-1fe, recovery, batch
perl: warning: Setting locale failed.
perl: warning: Please check that your locale settings:
	LANGUAGE = (unset),
	LC_ALL = (unset),
	LANG = "en_US.utf8"
    are supported and installed on your system.
perl: warning: Falling back to the standard locale ("C").
thread '<unnamed>' panicked at 'failed to run test after retry 5 times: test error at e2e_test/batch/./basic/logical_view.slt.part:7: query failed: db error: ERROR: QueryError: Catalog error: table or source not found: v1
```

Reply:
(Why GitHub is green when the test failed..